api/capi/include/nntrainer.h

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (C) 2019 Samsung Electronics Co., Ltd. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * @file nntrainer.h
+ * @date 02 April 2020
+ * @brief NNTrainer C-API Header.
+ *        This allows to construct and control NNTrainer Model.
+ * @see	https://github.com/nnstreamer/nntrainer
+ * @author Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+
+#ifndef __TIZEN_MACHINELEARNING_NNTRAINER_H__
+#define __TIZEN_MACHINELEARNING_NNTRAINER_H__
+
+#include <stdbool.h>
+#include <stddef.h>
+
+#include <tizen_error.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+/**
+ * @addtogroup CAPI_ML_NNTRAINER_MODULE
+ * @{
+ */
+
+/**
+ * @brief A handle of an NNTrainer model.
+ * @since_tizen 5.5
+ */
+typedef void *ml_nnmodel_h;
+
+/**
+ * @brief Enumeration for the error codes of NNTrainer.
+ * @since_tizen 5.5
+ */
+typedef enum {
+  ML_ERROR_NONE = TIZEN_ERROR_NONE,                                   /**< Success! */
+  ML_ERROR_INVALID_PARAMETER = TIZEN_ERROR_INVALID_PARAMETER,         /**< Invalid parameter */
+  ML_ERROR_UNKNOWN = TIZEN_ERROR_UNKNOWN,                             /**< Unknown error */
+  ML_ERROR_TIMED_OUT = TIZEN_ERROR_TIMED_OUT,                         /**< Time out */
+  ML_ERROR_NOT_SUPPORTED = TIZEN_ERROR_NOT_SUPPORTED,                 /**< The feature is not supported */
+  ML_ERROR_PERMISSION_DENIED = TIZEN_ERROR_PERMISSION_DENIED,         /**< Permission denied */
+  ML_ERROR_OUT_OF_MEMORY = TIZEN_ERROR_OUT_OF_MEMORY,                 /**< Out of memory (Since 6.0) */
+  ML_ERROR_CANNOT_ASSIGN_ADDRESS = TIZEN_ERROR_CANNOT_ASSIGN_ADDRESS, /**< CAnnot assign requested address */
+  ML_ERROR_BAD_ADDRESS = TIZEN_ERROR_BAD_ADDRESS,                     /**< Bad Address */
+} ml_error_e;
+
+/**
+ * @brief Constructs the neural network model.
+ * @details Use this function to create Neural Netowrk Model.
+ * @since_tizen 5.5
+ * @param[in] model The NNTrainer Model handler from the given description.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful
+ * @retval #ML_ERROR_CANNOT_ASSIGN_ADDRESS Cannot assign object.
```

Comment:
recommended to run checker before the ACR process.
Please see internal guideline for native API.
```
nntrainer.h:71: Question:
  Are you sure this function couldn't be given invalid parameter values? Please consider adding INVALID_PARAMETER error.

nntrainer.h:74: Question:
  Are you sure @brief is in third person (Gets ..., Sets ...)?
```

Reply:
Thanks. It will help me a lot.